File: src/main/java/com/kingmang/lazurite/runtime/values/LzrNull.java
package com.kingmang.lazurite.runtime.values;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class LzrNull implements LzrValue {


    @Nullable
    @Override
    public Object raw() {
        return null;
    }

    @Override
    public int asInt() {
        return 0;
    }

    @Override
    public double asNumber() {
        return 0;
    }

    @NotNull
    @Override
    public String asString() {
        return "null";
    }

    @NotNull
    @Override
    public int[] asArray() {
        return new int[0];
    }

    @Override
    public int type() {
        return 482862660;
    }

    @Override
    public int compareTo(@NotNull LzrValue o) {
        if (o.raw() == null) return 0;
        return -1;
    }

    @Override
    public String toString() {
        return asString();
    }
}

File: src/main/java/com/kingmang/lazurite/runtime/values/LzrNull.kt
package com.kingmang.lazurite.runtime.values

object LzrNull : LzrValue {
    override fun raw(): Any? =
        null

    override fun asInt(): Int =
        0

    override fun asNumber(): Double =
        0.0

    override fun asString(): String =
        "null"

    override fun asArray(): IntArray =
        IntArray(0)

    override fun type(): Int =
        482862660

    override fun compareTo(other: LzrValue): Int {
        other.raw() ?: return 0
        return -1
    }

    override fun toString(): String =
        this.asString()
}